cps/kni.c

```diff
@@ -979,6 +982,69 @@ moderror(int err)
 	}
 }
 
+#define SYS_MODULES_ATTR_PATH ("/sys/module/%s/%s")
+
+static int
+get_loaded_kmod_attr(const char *attr, char *val, size_t val_len)
+{
+	FILE *attr_file;
+	char path[128];
+	char line[64];
```

Comment:
Let's be generous here because `path` depends on a parameter and this function, `get_loaded_kmod_attr()`, should not assume that it is only going to be used to read small value. So let's give 256 bytes to `path` and 1024 to `line`.